pyfar/dsp/fractional_octave_smoothing.py

```
@@ -0,0 +1,317 @@
+    """
+    Converts two arrays of amplitude and phase into one array
+    of complex numbers in cartesian form.
+
+    Parameters
+    ----------
+    amplitude : ndarray
+        Array of amplitude values.
+    phase : ndarray
+        Array of phase values.
+
+    Returns
+    -------
+    ndarray
+        Array of same shape as input arrays with complex numbers.
+
+    Raises
+    ------
+    ValueError
+        Arrays must have same shape.
+    """
+    if not (amplitude.shape == phase.shape):
+        raise ValueError("Arrays must have same shapes.")
+    # Save input shape
+    input_shape = amplitude.shape
+    # Reshape for cmath rect to 1D arrays:
+    reshaped_amplitude = amplitude.reshape(1, -1)[0]
+    reshaped_phase = phase.reshape(1, -1)[0]
+    # Apply cmath rect on elements:
+    cartesian = np.array([cm.rect(a, p) for a, p in
+                          zip(reshaped_amplitude, reshaped_phase)])
+    return cartesian.reshape(input_shape)
+
+
+def frac_smooth_signal(signal, smoothing_width):
+    """
+    Method to smooth a given signal.
+    Takes the data of a given signal of shape (n, m), where n is number of
+    channels and m length of the signal.
+    Creates an object of class FractionalSmoothing to compute integration
+    limits and smoothing weights. Finally applies the weights on the input data
+    and returns smoothed signal.
+
+    Parameters
+    ----------
+    signal : Signal
+        Input signal to be smoothed
+    smoothing_width : float, int
+        Width of smoothing window relative to an octave
+
+    Returns
+    -------
+    Signal
+        Smoothed Signal
+
+    Raises
+    ------
+    TypeError
+        Input data must be of type Signal.
+    """
+    if not isinstance(signal, Signal):
+        raise TypeError("Input data must be of type Signal.")
+    # Create smoothing bject
+    obj = FractionalSmoothing(signal.n_bins, smoothing_width)
+    # Compute weights:
+    obj.calc_weights()
```

I did not check this - just posting:
For robustness, the weights should be removed in `FractionalSmoothing` if a related parameter changes (e.g. `n_bins`) and `apply` should either raise an error or calculate the weights itself, if they are missing.

---

The object can only be constructed with either the whole signal and the smoothing width or the number of frequency bins and the smoothing width. Thereby, the object should contain both parameters once the weights are computed. 
If `n_bins` or `smoothing_width` are changed through the setters, weights are updated.
`n_bins` and `smoothing_width` cannot be deleted.
Added a ValueError if `n_bins` or `smoothing_width` is None in `calc_integration_limits`